pkg/eval/builtin_fn_flow.go

```diff
@@ -50,7 +50,73 @@ import (
 
 // TODO(xiaq): Document "multi-error".
 
-// TODO(xiaq): Document "return", "break" and "continue".
+//elvdoc:fn break
+//
+// Raise the special "break" exception. Loops capture this exception and cause
+// the loop to terminate. Because it is an exception it can be caught by a
+// [`try`](language.html#exception-control-try) block. If not caught, either
+// implicitly by a loop or explicitly, it causes a failure like any other
+// uncaught exception.
+//
+// See the discussion about [flow commands and exceptions](language.html#exception-and-flow-commands)
+//
+// **Note**: You can create a `break` function and it will shadow the builtin
+// command. If you do so you should explicitly invoke the builtin. For example:
+//
+// ```elvish-transcript
+// > fn break []{ put break; builtin:break; put 'should not appear' }
+// > for x [a b c] { put $x; break }
+// ▶ a
+// ▶ break
+// ```
+
+//elvdoc:fn continue
+//
+// Raise the special "continue" exception. Loops capture this exception and
+// cause the loop to begin its next iteration. Because it is an exception it
+// can be caught by a [`try`](language.html#exception-control-try) block. If
+// not caught, either implicitly by a loop or explicitly, it causes a failure
+// like any other uncaught exception.
+//
+// See the discussion about [flow commands and exceptions](language.html#exception-and-flow-commands)
+//
+// **Note**: You can create a `continue` function and it will shadow the builtin
+// command. If you do so you should explicitly invoke the builtin. For example:
+//
+// ```elvish-transcript
+// > fn break []{ put continue; builtin:continue; put 'should not appear' }
+// > for x [a b c] { put $x; break }
+// ▶ a
+// ▶ continue
+// ▶ b
+// ▶ continue
+// ▶ c
+// ▶ continue
+// ```
+
+//elvdoc:fn return
+//
+// Raise the special "return" exception. Functions capture this exception and
+// cause the function to terminate. Because it is an exception it can be
```

Comment:
Same 3 comments as above.

Moreover, `return` exceptions are only captured by `fn`-defined functions, **not** anonymous functions; document that.